File: Modules/ServerTypes/Ticket/PriorityColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PriorityRow } from "./PriorityRow";

export interface PriorityColumns {
    Id: Column<PriorityRow>;
    DueTime: Column<PriorityRow>;
    Name: Column<PriorityRow>;
    IconName: Column<PriorityRow>;
    Color: Column<PriorityRow>;
    Description: Column<PriorityRow>;
}

export class PriorityColumns extends ColumnsBase<PriorityRow> {
    static readonly columnsKey = 'Ticket.Priority';
    static readonly Fields = fieldsProxy<PriorityColumns>();
}

File: Modules/Ticket/Priority/PriorityColumns.cs
namespace Ticketing.Ticket.Columns;

[ColumnsScript("Ticket.Priority")]
[BasedOnRow(typeof(PriorityRow), CheckNames = true)]
public class PriorityColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }    
    [EditLink]
    public string Name { get; set; }
    [PriorityFormatter]
    public string Color { get; set; }
    [Width(200)]
    public short DueTime { get; set; }
    public string Description { get; set; }
}